src/main/java/uk/ac/ebi/ena/webin/cli/manifest/processor/metadata/SampleProcessor.java

```diff
@@ -47,29 +62,111 @@ public Callback<Sample> getCallback() {
-          WebinCliMessage.SAMPLE_SERVICE_SYSTEM_ERROR.format(value));
-
+      if (isJsonValue(sampleValue)) {
+        uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample jsonSample = getJsonSample(sampleValue);
+        Submission submission = createSubmission(jsonSample);
+        WebinSubmission webinSubmission = createWebinSubmission(jsonSample,submission);
+        // There will be only one action for a sample
+        ActionType actionType = submission.getActions().get(0).getType();
+        if (actionType== ActionType.ADD || (actionType == ActionType.MODIFY && parameters.isSampleUpdate())) {
+          getSubmitService().doJsonSubmission(webinSubmission);
+        }
+        sampleValue = jsonSample.getAlias();
+      }
+      
+      Sample sample = getSample(sampleValue);
       fieldValue.setValue(sample.getBioSampleId());
       callback.notify(sample);
 
-    } catch (WebinCliException e) {
+    } catch (WebinCliException | JsonProcessingException e) {
       result.add(ValidationMessage.error(e));
     }
   }
+  
+  private WebinSubmission createWebinSubmission(uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample sample, Submission submission){
+    List<uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample> sampleList = new ArrayList();
+    sampleList.add(sample);
+    WebinSubmission webinSubmission = new WebinSubmission();
+    webinSubmission.setSubmission(submission);
+    webinSubmission.setSamples(sampleList);
+    return webinSubmission;
+  }
+  
+  private uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample getJsonSample(String sampleValue) throws JsonProcessingException {
+    uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample sampleFromUser = new ObjectMapper().readValue(sampleValue, uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample.class);
+    return sampleFromUser;
+  }
+  
+  private Submission createSubmission(uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample sample) {
+    Submission submission = new Submission();
+    List<Action> actions = new ArrayList<>();
+    Action action = isSample(sample) ? createAction(ActionType.MODIFY) : createAction(ActionType.ADD);
+    actions.add(action);
+    submission.setActions(actions);
+    return submission;
+  }
+
+  private Action createAction(ActionType type){
+    Action action=new Action();
+    action.setType(type);
+    return action;
+  }
+  
+  private Sample getSample(String sampleIdOrAlias){
+  
+    return  ExceptionUtils.executeWithRestExceptionHandling(() -> getSampleService().getSample(sampleIdOrAlias),
+            WebinCliMessage.SERVICE_AUTHENTICATION_ERROR.format(SampleService.SERVICE_NAME),
+            WebinCliMessage.SAMPLE_SERVICE_VALIDATION_ERROR.format(sampleIdOrAlias),
+            WebinCliMessage.SAMPLE_SERVICE_SYSTEM_ERROR.format(sampleIdOrAlias));
+  
+  }
+
+  private boolean isSample(uk.ac.ebi.ena.webin.xml.conversion.json.model.sample.Sample sampleFromUser){
```

Comment:
Add a comment:
//* Return true if the sample being submitters already exists in the submission account. */

 sampleFromUser -> sample